integration_tests/pkg/snap/session_test.go

```diff
@@ -89,25 +92,27 @@ func TestSnap(t *testing.T) {
 
 						tags := make(map[string]interface{})
 						tags["foo"] = "bar"
-						err := s.Start(tags)
+						handle, err := s.Launch(tags)
 
 						So(err, ShouldBeNil)
 
 						defer func() {
-							if s.IsRunning() {
-								err := s.Stop()
-								So(err, ShouldBeNil)
-							}
+							err := handle.Stop()
+							So(err, ShouldBeNil)
 						}()
 						Convey("Contacting snap to get the task status", func() {
-							So(s.IsRunning(), ShouldBeTrue)
+							So(handle.Status(), ShouldEqual, executor.RUNNING)
 
-							err = s.Wait()
+							terminated, err := handle.Wait(0)
 							So(err, ShouldBeNil)
+							So(terminated, ShouldBeTrue)
 
-							err = s.Stop()
+							err = handle.Stop()
```

Comment:
@skonefal - as a developer I would look at the test as example use case. It should not instruct user to run `Stop()` after `Wait(0)`. What about `Convey("It should be possible to call Stop() on a non-running task")`?